gateway-discovery-cm/src/main/java/org/apache/knox/gateway/topology/discovery/cm/monitor/PollingConfigurationAnalyzer.java

```diff
@@ -365,12 +375,32 @@ private DiscoveryApiClient getApiClient(final ServiceDiscoveryConfig discoveryCo
                                                clusterName,
                                                lastTimestamp);
     for (ApiEvent event : events) {
-      restartEvents.add(new RestartEvent(event));
+      List<ApiEventAttribute> attributes = event.getAttributes();
+      Map<String,Object> map = getAttributeMap(attributes);
+      addIfRelevantEvent(restartEvents, event, map);
     }
 
     return restartEvents;
   }
 
+  @SuppressWarnings("unchecked")
+  private void addIfRelevantEvent(List<RestartEvent> restartEvents, ApiEvent event, Map<String, Object> map) {
```

Comment:
Since the map is derived from the ApiEvent, addIfRelevantEvent should not need the attribute Map passed in as a param.

Reply:
I didn't see any point in changing it to reflect Start and Restart the are both "start" events. We can change to Start which is even a semantically meaningless change. I wanted to keep the map external rather than create it inside that method in case we need to add an additional use of the map later. We can move it in and refactor later if needed.

Reply:
Latest revision addresses the "restart" terminology and cleans up the method as you suggested offline. Thanks for the review!